sql/type.go

```diff
@@ -700,7 +700,19 @@ func (t datetimeT) Zero() interface{} {
 
 // DatetimeLayout is the layout of the MySQL date format in the representation
 // Go understands.
-const DatetimeLayout = "2006-01-02 15:04:05"
+const DatetimeLayout = "2006-01-02 15:04:05.999999"
+
+// DatetimeLayouts hold extra configurations allowed for parsing. It does
+// not have all the layouts supported by mysql. Missing are two digit year
+// versions of common cases and dates that use non common separators.
+//
+// https://github.com/MariaDB/server/blob/mysql-5.5.36/sql-common/my_time.c#L124
+var DatetimeLayouts = []string{
+	"2006-01-02",
```

Comment:
I would include the main constant in this slice, then just loop over it a single time